{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# ALZML\n",
    "\n",
    "The goal of this project is to use publically available datasets construct a classifier to determine the features that confer Alzheimer's Disease. To do so we are combining SNP data with metadata to generate a feature rich dataset. \n",
    "\n",
    "First we will identify ROIs from SNP data that are known to be involved in AD. From there, adjacent features will be extracted to serve as a feature profile of the ROI. Once all ROIs have been profiled we will then run a heirarchical clustering algorithm to determine distinguishing features of these ROIs.\n",
    "\n",
    "From these features we are hoping to construct an HMM that will be able to identify some of the underrepresented variants that may confer AD. \n",
    "\n",
    "Data are sourced from:\n",
    "- gnomad\n",
    "- IGAP\n",
    "- TBI Study\n",
    "- Zou eGWAS\n",
    "- Mayo eGWAS\n",
    "- NG00061\n",
    "- NG00039\n",
    "\n",
    "IGAP data will be used as the determinant for a ROI. A +/- 5kb region from the location of the SNP will serve as an anchor to gather local features using bedtools in order to retain positional information. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Only run bash block if files need to be parsed properly (shouldn't happen)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "#head -n 10 ng00061/WES_release3AtlasOnly_vep80_most_severe_consequence_per_gene.txt\n",
    "# These have different numbers of columns of course. Need to fix this maybe? Will only use the coordinates from this\n",
    "# for now, this is the primary key to link all the other datasets.\n",
    "#head -n 10 ng00061/WES_release3AtlasOnly_rolling_flat_annotation.txt\n",
    "#cut -f1-5 WGS_v1_rolling_flat_annotation.txt > WGS_v1_rolling_flat_annotation.pos_only.txt\n",
    "\n",
    "#head -n 10 ../IGAP_summary_statistics/IGAP_stage_1_2_combined.txt\n",
    "\n",
    "# I will join all of the chromosomes into one file.\n",
    "#cat /home/twaddlac/Hackthan_AD/ng00039/pvalue_only/metaanalysis/pvalueonly_METAANALYSIS1_chr*.TBL | perl -pe 's/^(\\d+)-(\\d+)/$1\\t$2/g' > pvalue_only.tsv\n",
    "#head -n 10 ../pvalue_only.tsv\n",
    "\n",
    "#I'm not sure what the difference is between these but I am assuming we get the inverse of controls here.\n",
    "# We will use the controls for a confusion matrix\n",
    "#ls ../Mayo_eGWAS/\n",
    "#head ../Mayo_eGWAS/Hap300_CER_All.txt\n",
    "\n",
    "# Only going to use the coordinates of the annotation files since they have mismatching columns.\n",
    "##### Only run this once #####\n",
    "#cut -f1-5 ../NG00061/WGS_v1_rolling_flat_annotation.txt > ../NG00061/WGS_v1_rolling_flat_annotation.pos_only.txt\n",
    "\n",
    "## only need to run once\n",
    "#cat <(grep -m1 '^Marker' /home/twaddlac/Hackthan_AD/ng00039/pvalue_only/pvalue/pvalueonly_METAANALYSIS1_chr10.TBL) <(cat /home/twaddlac/Hackthan_AD/ng00039/pvalue_only/pvalue/*TBL | perl -pe 's/(\\d+)-(\\d+)/$1\\t$2/g'| grep -v '^Marker') > /home/twaddlac/Hackthan_AD/ng00039/pvalue_only/pvalue/pvalue.tsv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## IGAP Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "igap1 = pd.read_csv('../IGAP_summary_statistics/IGAP_stage_1.txt', sep='\\t')\n",
    "igap12 = pd.read_csv('../IGAP_summary_statistics/IGAP_stage_1_2_combined.txt', sep='\\t')\n",
    "igap1.rename(columns={\n",
    "    'Chromosome':'chr',\n",
    "    'Position':'pos'\n",
    "}, inplace=True)\n",
    "igap12.rename(columns={\n",
    "    'Chromosome':'chr',\n",
    "    'Position':'pos'\n",
    "}, inplace=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## NG00061"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "anno = pd.read_csv('../NG00061/WGS_v1_rolling_flat_annotation.pos_only.txt', sep='\\t', header=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "conseq = pd.read_csv('../NG00061/WGS_v1_vep80_most_severe_consequence_per_gene.txt', sep='\\t', header=0)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## NG00039"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] File b'../ng00039/pvalue_only/pvalue/pvalue.tsv' does not exist: b'../ng00039/pvalue_only/pvalue/pvalue.tsv'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-5-f3cd1db8f5e1>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mpvalue\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread_csv\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'../ng00039/pvalue_only/pvalue/pvalue.tsv'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msep\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'\\t'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mheader\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mindex_col\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      2\u001b[0m \u001b[0;31m# pvalue.columns = ['chr','pos','allele1','allele2','pvalue']\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/local/bin/anaconda/lib/python3.7/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36mparser_f\u001b[0;34m(filepath_or_buffer, sep, delimiter, header, names, index_col, usecols, squeeze, prefix, mangle_dupe_cols, dtype, engine, converters, true_values, false_values, skipinitialspace, skiprows, skipfooter, nrows, na_values, keep_default_na, na_filter, verbose, skip_blank_lines, parse_dates, infer_datetime_format, keep_date_col, date_parser, dayfirst, iterator, chunksize, compression, thousands, decimal, lineterminator, quotechar, quoting, doublequote, escapechar, comment, encoding, dialect, tupleize_cols, error_bad_lines, warn_bad_lines, delim_whitespace, low_memory, memory_map, float_precision)\u001b[0m\n\u001b[1;32m    700\u001b[0m                     skip_blank_lines=skip_blank_lines)\n\u001b[1;32m    701\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 702\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0m_read\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfilepath_or_buffer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkwds\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    703\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    704\u001b[0m     \u001b[0mparser_f\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__name__\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mname\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/local/bin/anaconda/lib/python3.7/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36m_read\u001b[0;34m(filepath_or_buffer, kwds)\u001b[0m\n\u001b[1;32m    427\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    428\u001b[0m     \u001b[0;31m# Create the parser.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 429\u001b[0;31m     \u001b[0mparser\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mTextFileReader\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfilepath_or_buffer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwds\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    430\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    431\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0mchunksize\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0miterator\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/local/bin/anaconda/lib/python3.7/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, f, engine, **kwds)\u001b[0m\n\u001b[1;32m    893\u001b[0m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0moptions\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'has_index_names'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mkwds\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'has_index_names'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    894\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 895\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_make_engine\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mengine\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    896\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    897\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mclose\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/local/bin/anaconda/lib/python3.7/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36m_make_engine\u001b[0;34m(self, engine)\u001b[0m\n\u001b[1;32m   1120\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_make_engine\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mengine\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'c'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1121\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mengine\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m'c'\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1122\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mCParserWrapper\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0moptions\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1123\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1124\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mengine\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m'python'\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/local/bin/anaconda/lib/python3.7/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, src, **kwds)\u001b[0m\n\u001b[1;32m   1851\u001b[0m         \u001b[0mkwds\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'usecols'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0musecols\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1852\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1853\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_reader\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mparsers\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mTextReader\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msrc\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwds\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1854\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0munnamed_cols\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_reader\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0munnamed_cols\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1855\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/parsers.pyx\u001b[0m in \u001b[0;36mpandas._libs.parsers.TextReader.__cinit__\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/parsers.pyx\u001b[0m in \u001b[0;36mpandas._libs.parsers.TextReader._setup_parser_source\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;31mFileNotFoundError\u001b[0m: [Errno 2] File b'../ng00039/pvalue_only/pvalue/pvalue.tsv' does not exist: b'../ng00039/pvalue_only/pvalue/pvalue.tsv'"
     ]
    }
   ],
   "source": [
    "pvalue = pd.read_csv('../ng00039/pvalue_only/pvalue/pvalue.tsv', sep='\\t', header=0, index_col=False)\n",
    "# pvalue.columns = ['chr','pos','allele1','allele2','pvalue']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Mayo_eGWAS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "hapCerAd = pd.read_csv('/home/twaddlac/Hackthan_AD/Mayo_eGWAS/Hap300_CER_AD.txt', sep='\\t', header=0)\n",
    "hapTxAd = pd.read_csv('/home/twaddlac/Hackthan_AD/Mayo_eGWAS/Hap300_TX_AD.txt', sep='\\t', header=0)\n",
    "hapmapCerAd = pd.read_csv('/home/twaddlac/Hackthan_AD/Mayo_eGWAS/HapMap2_CER_AD.txt', sep='\\t', header=0)\n",
    "hapmapTxAd = pd.read_csv('/home/twaddlac/Hackthan_AD/Mayo_eGWAS/HapMap2_TX_AD.txt', sep='\\t', header=0)\n",
    "\n",
    "hapCerAd.rename(columns={'CHR':'chr','BP':'pos'}, inplace=True)\n",
    "hapTxAd.rename(columns={'CHR':'chr','BP':'pos'}, inplace=True)\n",
    "hapmapCerAd.rename(columns={'CHR':'chr','BP':'pos'}, inplace=True)\n",
    "hapmapTxAd.rename(columns={'CHR':'chr','BP':'pos'}, inplace=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## TBI Study Expression Data\n",
    "There's a lot more data for this dataset but we can import it later."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "ge = pd.read_csv('../TBI_study/gene_expression_matrix_2016-03-03/fpkm_table_normalized.csv', sep=',', header=0)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## gnomad data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "gnomad = pd.read_csv('../IGAP_summary_statistics/gnomad_gwas_intersect.txt', )"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# CRAVAT Data\n",
    "CRAVAT dumps a sqllite3 file that I've read in. The only interesting tables seem to be variants and genes.\n",
    "\n",
    "The genes that are implicated in AD are seen below:\n",
    "Chromosome 11       PICAL\n",
    "Chromosome 2        BIN1\n",
    "Chromosome 19      CD33\n",
    "Chromosome 1        CR1\n",
    "Chromosome 7       EPHA1\n",
    "Chromosome 6       TREM2\n",
    "Chromosome 19.     ABCA7\n",
    "Chromosome 11      SORL\n",
    "Chromosome 12.    ADAM10\n",
    "Chromosome     2.     ADAM17\n",
    "Chromosome 7    AKAP9\n",
    "Chromosome 4    UNC5C\n",
    "Chromosome 19    APOE"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "It seems like clustering mulitple datatypes can be done as described in this article:\n",
    "https://towardsdatascience.com/clustering-on-mixed-type-data-8bbd0a2569c3\n",
    "\n",
    "Let's try it out!\n",
    "\n",
    "The first step is to convert all of the categorical data to some numeric representation. I found a good post on how to do so [here](https://pbpython.com/categorical-encoding.html)\n",
    "\n",
    "We will then also make some assumtions about the data to help reduce the number of features (columns) represented in the dataframe. Those being:\n",
    "If there is `none` for the transcript ID then we will assume it is in an exon. We can therefore drop the gene.\n",
    "\n",
    "TO DROP:\n",
    "base__chrom\n",
    "base__pos\n",
    "base__note\n",
    "base__hugo\n",
    "base__all_mappings\n",
    "vest__all_results\n",
    "vest__hugo\n",
    "vest__transcript\n",
    "base__uid\n",
    "denovo__PubmedId\n",
    "denovo__Validation\n",
    "denovo__PrimaryPhenotype\n",
    "cosmic__protein_change\n",
    "cosmic__transcript\n",
    "cosmic__variant_count\n",
    "cosmic__cosmic_id\n",
    "clinvar__sig\n",
    "clinvar__id\n",
    "clinvar__rev_stat\n",
    "clinvar__disease_refs\n",
    "clinvar__disease_names\n",
    "gtex__gtex_gene\n",
    "gtex__gtex_tissue\n",
    "uk10k_cohort__uk10k_twins_ac\n",
    "uk10k_cohort__uk10k_alspac_ac\n",
    "uk10k_cohort__uk10k_ac\n",
    "vest__pval\n",
    "vest__score\n",
    "tagsampler__numsample\n",
    "tagsampler__samples\n",
    "tagsampler__tags\n",
    "repeat__repeatname\n",
    "dbsnp__snp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>base__uid</th>\n",
       "      <th>base__chrom</th>\n",
       "      <th>base__pos</th>\n",
       "      <th>base__ref_base</th>\n",
       "      <th>base__alt_base</th>\n",
       "      <th>base__note</th>\n",
       "      <th>base__coding</th>\n",
       "      <th>base__hugo</th>\n",
       "      <th>base__transcript</th>\n",
       "      <th>base__so</th>\n",
       "      <th>base__achange</th>\n",
       "      <th>base__all_mappings</th>\n",
       "      <th>clinvar__sig</th>\n",
       "      <th>clinvar__disease_refs</th>\n",
       "      <th>clinvar__disease_names</th>\n",
       "      <th>clinvar__rev_stat</th>\n",
       "      <th>clinvar__id</th>\n",
       "      <th>cosmic__cosmic_id</th>\n",
       "      <th>cosmic__variant_count_tissue</th>\n",
       "      <th>cosmic__variant_count</th>\n",
       "      <th>cosmic__transcript</th>\n",
       "      <th>cosmic__protein_change</th>\n",
       "      <th>dbsnp__snp</th>\n",
       "      <th>denovo__PubmedId</th>\n",
       "      <th>denovo__PrimaryPhenotype</th>\n",
       "      <th>denovo__Validation</th>\n",
       "      <th>esp6500__ea_pop_af</th>\n",
       "      <th>esp6500__aa_pop_af</th>\n",
       "      <th>gnomad__af</th>\n",
       "      <th>gnomad__af_afr</th>\n",
       "      <th>gnomad__af_amr</th>\n",
       "      <th>gnomad__af_asj</th>\n",
       "      <th>gnomad__af_eas</th>\n",
       "      <th>gnomad__af_fin</th>\n",
       "      <th>gnomad__af_nfe</th>\n",
       "      <th>gnomad__af_oth</th>\n",
       "      <th>gnomad__af_sas</th>\n",
       "      <th>gtex__gtex_gene</th>\n",
       "      <th>gtex__gtex_tissue</th>\n",
       "      <th>repeat__repeatclass</th>\n",
       "      <th>repeat__repeatfamily</th>\n",
       "      <th>repeat__repeatname</th>\n",
       "      <th>thousandgenomes__af</th>\n",
       "      <th>thousandgenomes__afr_af</th>\n",
       "      <th>thousandgenomes__amr_af</th>\n",
       "      <th>thousandgenomes__eas_af</th>\n",
       "      <th>thousandgenomes__eur_af</th>\n",
       "      <th>thousandgenomes__sas_af</th>\n",
       "      <th>thousandgenomes__chb_af</th>\n",
       "      <th>thousandgenomes__jpt_af</th>\n",
       "      <th>thousandgenomes__chs_af</th>\n",
       "      <th>thousandgenomes__cdx_af</th>\n",
       "      <th>thousandgenomes__khv_af</th>\n",
       "      <th>thousandgenomes__ceu_af</th>\n",
       "      <th>thousandgenomes__tsi_af</th>\n",
       "      <th>thousandgenomes__fin_af</th>\n",
       "      <th>thousandgenomes__gbr_af</th>\n",
       "      <th>thousandgenomes__ibs_af</th>\n",
       "      <th>thousandgenomes__yri_af</th>\n",
       "      <th>thousandgenomes__lwk_af</th>\n",
       "      <th>thousandgenomes__gwd_af</th>\n",
       "      <th>thousandgenomes__msl_af</th>\n",
       "      <th>thousandgenomes__esn_af</th>\n",
       "      <th>thousandgenomes__asw_af</th>\n",
       "      <th>thousandgenomes__acb_af</th>\n",
       "      <th>thousandgenomes__mxl_af</th>\n",
       "      <th>thousandgenomes__pur_af</th>\n",
       "      <th>thousandgenomes__clm_af</th>\n",
       "      <th>thousandgenomes__pel_af</th>\n",
       "      <th>thousandgenomes__gih_af</th>\n",
       "      <th>thousandgenomes__pjl_af</th>\n",
       "      <th>thousandgenomes__beb_af</th>\n",
       "      <th>thousandgenomes__stu_af</th>\n",
       "      <th>thousandgenomes__itu_af</th>\n",
       "      <th>uk10k_cohort__uk10k_twins_ac</th>\n",
       "      <th>uk10k_cohort__uk10k_twins_af</th>\n",
       "      <th>uk10k_cohort__uk10k_alspac_ac</th>\n",
       "      <th>uk10k_cohort__uk10k_alspac_af</th>\n",
       "      <th>uk10k_cohort__uk10k_ac</th>\n",
       "      <th>uk10k_cohort__uk10k_af</th>\n",
       "      <th>vest__transcript</th>\n",
       "      <th>vest__score</th>\n",
       "      <th>vest__pval</th>\n",
       "      <th>vest__all_results</th>\n",
       "      <th>vest__hugo</th>\n",
       "      <th>vista_enhancer__element</th>\n",
       "      <th>vista_enhancer__features</th>\n",
       "      <th>tagsampler__numsample</th>\n",
       "      <th>tagsampler__samples</th>\n",
       "      <th>tagsampler__tags</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>chr19</td>\n",
       "      <td>39418764</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>None</td>\n",
       "      <td>Y</td>\n",
       "      <td>PLEKHG2</td>\n",
       "      <td>ENST00000425673.5</td>\n",
       "      <td>STG</td>\n",
       "      <td>R372*</td>\n",
       "      <td>{\"PLEKHG2\":[[null,\"R372*\",\"STG\",\"ENST00000409797.6\",\"C1114T\"],[\"Q9H7P9\",\"R372*\",\"STG\",\"ENST00000425673.5\",\"C1114T\"],[\"E7ESZ3\",\"R313*\",\"STG\",\"ENST00000458508.6\",\"C937T\"]]}</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>rs774367341</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000028</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000099</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000044</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000033</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000458508.6</td>\n",
       "      <td>0.441</td>\n",
       "      <td>0.29194</td>\n",
       "      <td>ENST00000409797.6(0.401:0.3212),ENST00000425673.5(0.435:0.29606),*ENST00000458508.6(0.441:0.29194)</td>\n",
       "      <td>PLEKHG2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>chr19</td>\n",
       "      <td>39418783</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>None</td>\n",
       "      <td>Y</td>\n",
       "      <td>PLEKHG2</td>\n",
       "      <td>ENST00000425673.5</td>\n",
       "      <td>MIS</td>\n",
       "      <td>K378R</td>\n",
       "      <td>{\"PLEKHG2\":[[null,\"K378R\",\"MIS\",\"ENST00000409797.6\",\"A1133G\"],[\"Q9H7P9\",\"K378R\",\"MIS\",\"ENST00000425673.5\",\"A1133G\"],[\"E7ESZ3\",\"K319R\",\"MIS\",\"ENST00000458508.6\",\"A956G\"]]}</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>rs1429162441</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000064</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000130</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000409797.6</td>\n",
       "      <td>0.278</td>\n",
       "      <td>0.38063</td>\n",
       "      <td>*ENST00000409797.6(0.278:0.38063),ENST00000425673.5(0.236:0.46169),ENST00000458508.6(0.257:0.41939)</td>\n",
       "      <td>PLEKHG2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>chr19</td>\n",
       "      <td>39418784</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>None</td>\n",
       "      <td>Y</td>\n",
       "      <td>PLEKHG2</td>\n",
       "      <td>ENST00000425673.5</td>\n",
       "      <td>MIS</td>\n",
       "      <td>K378N</td>\n",
       "      <td>{\"PLEKHG2\":[[null,\"K378N\",\"MIS\",\"ENST00000409797.6\",\"G1134T\"],[\"Q9H7P9\",\"K378N\",\"MIS\",\"ENST00000425673.5\",\"G1134T\"],[\"E7ESZ3\",\"K319N\",\"MIS\",\"ENST00000458508.6\",\"G957T\"]]}</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>rs773266798</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000032</td>\n",
       "      <td>0.000115</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000409797.6</td>\n",
       "      <td>0.564</td>\n",
       "      <td>0.14756</td>\n",
       "      <td>*ENST00000409797.6(0.564:0.14756),ENST00000425673.5(0.564:0.14756),ENST00000458508.6(0.532:0.16962)</td>\n",
       "      <td>PLEKHG2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>chr19</td>\n",
       "      <td>39418787</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>None</td>\n",
       "      <td>Y</td>\n",
       "      <td>PLEKHG2</td>\n",
       "      <td>ENST00000425673.5</td>\n",
       "      <td>SYN</td>\n",
       "      <td>V379V</td>\n",
       "      <td>{\"PLEKHG2\":[[null,\"V379V\",\"SYN\",\"ENST00000409797.6\",\"G1137A\"],[\"Q9H7P9\",\"V379V\",\"SYN\",\"ENST00000425673.5\",\"G1137A\"],[\"E7ESZ3\",\"V320V\",\"SYN\",\"ENST00000458508.6\",\"G960A\"]]}</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>rs766212173</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>0.000120</td>\n",
       "      <td>0.000028</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>chr19</td>\n",
       "      <td>39418797</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>None</td>\n",
       "      <td>Y</td>\n",
       "      <td>PLEKHG2</td>\n",
       "      <td>ENST00000425673.5</td>\n",
       "      <td>MIS</td>\n",
       "      <td>T383A</td>\n",
       "      <td>{\"PLEKHG2\":[[null,\"T383A\",\"MIS\",\"ENST00000409797.6\",\"A1147G\"],[\"Q9H7P9\",\"T383A\",\"MIS\",\"ENST00000425673.5\",\"A1147G\"],[\"E7ESZ3\",\"T324A\",\"MIS\",\"ENST00000458508.6\",\"A970G\"]]}</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>rs1301557927</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000032</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000642</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000409797.6</td>\n",
       "      <td>0.530</td>\n",
       "      <td>0.17093</td>\n",
       "      <td>*ENST00000409797.6(0.53:0.17093),ENST00000425673.5(0.423:0.22741),ENST00000458508.6(0.523:0.17377)</td>\n",
       "      <td>PLEKHG2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   base__uid base__chrom  base__pos base__ref_base base__alt_base base__note  \\\n",
       "0          1       chr19   39418764              C              T       None   \n",
       "1          2       chr19   39418783              A              G       None   \n",
       "2          3       chr19   39418784              G              T       None   \n",
       "3          4       chr19   39418787              G              A       None   \n",
       "4          5       chr19   39418797              A              G       None   \n",
       "\n",
       "  base__coding base__hugo   base__transcript base__so base__achange  \\\n",
       "0            Y    PLEKHG2  ENST00000425673.5      STG         R372*   \n",
       "1            Y    PLEKHG2  ENST00000425673.5      MIS         K378R   \n",
       "2            Y    PLEKHG2  ENST00000425673.5      MIS         K378N   \n",
       "3            Y    PLEKHG2  ENST00000425673.5      SYN         V379V   \n",
       "4            Y    PLEKHG2  ENST00000425673.5      MIS         T383A   \n",
       "\n",
       "                                                                                                                                                           base__all_mappings  \\\n",
       "0  {\"PLEKHG2\":[[null,\"R372*\",\"STG\",\"ENST00000409797.6\",\"C1114T\"],[\"Q9H7P9\",\"R372*\",\"STG\",\"ENST00000425673.5\",\"C1114T\"],[\"E7ESZ3\",\"R313*\",\"STG\",\"ENST00000458508.6\",\"C937T\"]]}   \n",
       "1  {\"PLEKHG2\":[[null,\"K378R\",\"MIS\",\"ENST00000409797.6\",\"A1133G\"],[\"Q9H7P9\",\"K378R\",\"MIS\",\"ENST00000425673.5\",\"A1133G\"],[\"E7ESZ3\",\"K319R\",\"MIS\",\"ENST00000458508.6\",\"A956G\"]]}   \n",
       "2  {\"PLEKHG2\":[[null,\"K378N\",\"MIS\",\"ENST00000409797.6\",\"G1134T\"],[\"Q9H7P9\",\"K378N\",\"MIS\",\"ENST00000425673.5\",\"G1134T\"],[\"E7ESZ3\",\"K319N\",\"MIS\",\"ENST00000458508.6\",\"G957T\"]]}   \n",
       "3  {\"PLEKHG2\":[[null,\"V379V\",\"SYN\",\"ENST00000409797.6\",\"G1137A\"],[\"Q9H7P9\",\"V379V\",\"SYN\",\"ENST00000425673.5\",\"G1137A\"],[\"E7ESZ3\",\"V320V\",\"SYN\",\"ENST00000458508.6\",\"G960A\"]]}   \n",
       "4  {\"PLEKHG2\":[[null,\"T383A\",\"MIS\",\"ENST00000409797.6\",\"A1147G\"],[\"Q9H7P9\",\"T383A\",\"MIS\",\"ENST00000425673.5\",\"A1147G\"],[\"E7ESZ3\",\"T324A\",\"MIS\",\"ENST00000458508.6\",\"A970G\"]]}   \n",
       "\n",
       "  clinvar__sig clinvar__disease_refs clinvar__disease_names clinvar__rev_stat  \\\n",
       "0         None                  None                   None              None   \n",
       "1         None                  None                   None              None   \n",
       "2         None                  None                   None              None   \n",
       "3         None                  None                   None              None   \n",
       "4         None                  None                   None              None   \n",
       "\n",
       "  clinvar__id cosmic__cosmic_id cosmic__variant_count_tissue  \\\n",
       "0        None              None                         None   \n",
       "1        None              None                         None   \n",
       "2        None              None                         None   \n",
       "3        None              None                         None   \n",
       "4        None              None                         None   \n",
       "\n",
       "   cosmic__variant_count cosmic__transcript cosmic__protein_change  \\\n",
       "0                    NaN               None                   None   \n",
       "1                    NaN               None                   None   \n",
       "2                    NaN               None                   None   \n",
       "3                    NaN               None                   None   \n",
       "4                    NaN               None                   None   \n",
       "\n",
       "     dbsnp__snp denovo__PubmedId denovo__PrimaryPhenotype denovo__Validation  \\\n",
       "0   rs774367341             None                     None               None   \n",
       "1  rs1429162441             None                     None               None   \n",
       "2   rs773266798             None                     None               None   \n",
       "3   rs766212173             None                     None               None   \n",
       "4  rs1301557927             None                     None               None   \n",
       "\n",
       "   esp6500__ea_pop_af  esp6500__aa_pop_af  gnomad__af  gnomad__af_afr  \\\n",
       "0                 NaN                 NaN    0.000028             NaN   \n",
       "1                 NaN                 NaN    0.000064             NaN   \n",
       "2                 NaN                 NaN    0.000032        0.000115   \n",
       "3                 NaN                 NaN    0.000014        0.000120   \n",
       "4                 NaN                 NaN    0.000032             NaN   \n",
       "\n",
       "   gnomad__af_amr  gnomad__af_asj  gnomad__af_eas  gnomad__af_fin  \\\n",
       "0             NaN        0.000099             NaN             NaN   \n",
       "1             NaN             NaN             NaN             NaN   \n",
       "2             NaN             NaN             NaN             NaN   \n",
       "3        0.000028             NaN             NaN             NaN   \n",
       "4             NaN             NaN        0.000642             NaN   \n",
       "\n",
       "   gnomad__af_nfe  gnomad__af_oth  gnomad__af_sas gtex__gtex_gene  \\\n",
       "0        0.000044             NaN        0.000033            None   \n",
       "1        0.000130             NaN             NaN            None   \n",
       "2             NaN             NaN             NaN            None   \n",
       "3             NaN             NaN             NaN            None   \n",
       "4             NaN             NaN             NaN            None   \n",
       "\n",
       "  gtex__gtex_tissue repeat__repeatclass repeat__repeatfamily  \\\n",
       "0              None                None                 None   \n",
       "1              None                None                 None   \n",
       "2              None                None                 None   \n",
       "3              None                None                 None   \n",
       "4              None                None                 None   \n",
       "\n",
       "  repeat__repeatname thousandgenomes__af thousandgenomes__afr_af  \\\n",
       "0               None                None                    None   \n",
       "1               None                None                    None   \n",
       "2               None                None                    None   \n",
       "3               None                None                    None   \n",
       "4               None                None                    None   \n",
       "\n",
       "  thousandgenomes__amr_af thousandgenomes__eas_af thousandgenomes__eur_af  \\\n",
       "0                    None                    None                    None   \n",
       "1                    None                    None                    None   \n",
       "2                    None                    None                    None   \n",
       "3                    None                    None                    None   \n",
       "4                    None                    None                    None   \n",
       "\n",
       "  thousandgenomes__sas_af thousandgenomes__chb_af thousandgenomes__jpt_af  \\\n",
       "0                    None                    None                    None   \n",
       "1                    None                    None                    None   \n",
       "2                    None                    None                    None   \n",
       "3                    None                    None                    None   \n",
       "4                    None                    None                    None   \n",
       "\n",
       "  thousandgenomes__chs_af thousandgenomes__cdx_af thousandgenomes__khv_af  \\\n",
       "0                    None                    None                    None   \n",
       "1                    None                    None                    None   \n",
       "2                    None                    None                    None   \n",
       "3                    None                    None                    None   \n",
       "4                    None                    None                    None   \n",
       "\n",
       "  thousandgenomes__ceu_af thousandgenomes__tsi_af thousandgenomes__fin_af  \\\n",
       "0                    None                    None                    None   \n",
       "1                    None                    None                    None   \n",
       "2                    None                    None                    None   \n",
       "3                    None                    None                    None   \n",
       "4                    None                    None                    None   \n",
       "\n",
       "  thousandgenomes__gbr_af thousandgenomes__ibs_af thousandgenomes__yri_af  \\\n",
       "0                    None                    None                    None   \n",
       "1                    None                    None                    None   \n",
       "2                    None                    None                    None   \n",
       "3                    None                    None                    None   \n",
       "4                    None                    None                    None   \n",
       "\n",
       "  thousandgenomes__lwk_af thousandgenomes__gwd_af thousandgenomes__msl_af  \\\n",
       "0                    None                    None                    None   \n",
       "1                    None                    None                    None   \n",
       "2                    None                    None                    None   \n",
       "3                    None                    None                    None   \n",
       "4                    None                    None                    None   \n",
       "\n",
       "  thousandgenomes__esn_af thousandgenomes__asw_af thousandgenomes__acb_af  \\\n",
       "0                    None                    None                    None   \n",
       "1                    None                    None                    None   \n",
       "2                    None                    None                    None   \n",
       "3                    None                    None                    None   \n",
       "4                    None                    None                    None   \n",
       "\n",
       "  thousandgenomes__mxl_af thousandgenomes__pur_af thousandgenomes__clm_af  \\\n",
       "0                    None                    None                    None   \n",
       "1                    None                    None                    None   \n",
       "2                    None                    None                    None   \n",
       "3                    None                    None                    None   \n",
       "4                    None                    None                    None   \n",
       "\n",
       "  thousandgenomes__pel_af thousandgenomes__gih_af thousandgenomes__pjl_af  \\\n",
       "0                    None                    None                    None   \n",
       "1                    None                    None                    None   \n",
       "2                    None                    None                    None   \n",
       "3                    None                    None                    None   \n",
       "4                    None                    None                    None   \n",
       "\n",
       "  thousandgenomes__beb_af thousandgenomes__stu_af thousandgenomes__itu_af  \\\n",
       "0                    None                    None                    None   \n",
       "1                    None                    None                    None   \n",
       "2                    None                    None                    None   \n",
       "3                    None                    None                    None   \n",
       "4                    None                    None                    None   \n",
       "\n",
       "   uk10k_cohort__uk10k_twins_ac  uk10k_cohort__uk10k_twins_af  \\\n",
       "0                           NaN                           NaN   \n",
       "1                           NaN                           NaN   \n",
       "2                           NaN                           NaN   \n",
       "3                           NaN                           NaN   \n",
       "4                           NaN                           NaN   \n",
       "\n",
       "   uk10k_cohort__uk10k_alspac_ac  uk10k_cohort__uk10k_alspac_af  \\\n",
       "0                            NaN                            NaN   \n",
       "1                            NaN                            NaN   \n",
       "2                            NaN                            NaN   \n",
       "3                            NaN                            NaN   \n",
       "4                            NaN                            NaN   \n",
       "\n",
       "   uk10k_cohort__uk10k_ac  uk10k_cohort__uk10k_af   vest__transcript  \\\n",
       "0                     NaN                     NaN  ENST00000458508.6   \n",
       "1                     NaN                     NaN  ENST00000409797.6   \n",
       "2                     NaN                     NaN  ENST00000409797.6   \n",
       "3                     NaN                     NaN               None   \n",
       "4                     NaN                     NaN  ENST00000409797.6   \n",
       "\n",
       "   vest__score  vest__pval  \\\n",
       "0        0.441     0.29194   \n",
       "1        0.278     0.38063   \n",
       "2        0.564     0.14756   \n",
       "3          NaN         NaN   \n",
       "4        0.530     0.17093   \n",
       "\n",
       "                                                                                     vest__all_results  \\\n",
       "0   ENST00000409797.6(0.401:0.3212),ENST00000425673.5(0.435:0.29606),*ENST00000458508.6(0.441:0.29194)   \n",
       "1  *ENST00000409797.6(0.278:0.38063),ENST00000425673.5(0.236:0.46169),ENST00000458508.6(0.257:0.41939)   \n",
       "2  *ENST00000409797.6(0.564:0.14756),ENST00000425673.5(0.564:0.14756),ENST00000458508.6(0.532:0.16962)   \n",
       "3                                                                                                 None   \n",
       "4   *ENST00000409797.6(0.53:0.17093),ENST00000425673.5(0.423:0.22741),ENST00000458508.6(0.523:0.17377)   \n",
       "\n",
       "  vest__hugo vista_enhancer__element vista_enhancer__features  \\\n",
       "0    PLEKHG2                    None                     None   \n",
       "1    PLEKHG2                    None                     None   \n",
       "2    PLEKHG2                    None                     None   \n",
       "3       None                    None                     None   \n",
       "4    PLEKHG2                    None                     None   \n",
       "\n",
       "   tagsampler__numsample tagsampler__samples tagsampler__tags  \n",
       "0                      0                                       \n",
       "1                      0                                       \n",
       "2                      0                                       \n",
       "3                      0                                       \n",
       "4                      0                                       "
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.options.display.max_columns = 500\n",
    "pd.options.display.max_colwidth = 1000\n",
    "variant.head()\n",
    "# variant['base__all_mappings']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 215,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/local/bin/anaconda/lib/python3.7/site-packages/ipykernel_launcher.py:77: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "/usr/local/bin/anaconda/lib/python3.7/site-packages/ipykernel_launcher.py:97: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "/usr/local/bin/anaconda/lib/python3.7/site-packages/ipykernel_launcher.py:98: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>base__coding_Y</th>\n",
       "      <th>base__transcript_1</th>\n",
       "      <th>base__so_2KD</th>\n",
       "      <th>base__so_2KU</th>\n",
       "      <th>base__so_CSS</th>\n",
       "      <th>base__so_FSD</th>\n",
       "      <th>base__so_IND</th>\n",
       "      <th>base__so_INT</th>\n",
       "      <th>base__so_MIS</th>\n",
       "      <th>base__so_SPL</th>\n",
       "      <th>base__so_STG</th>\n",
       "      <th>base__so_STL</th>\n",
       "      <th>base__so_SYN</th>\n",
       "      <th>base__so_UNK</th>\n",
       "      <th>base__so_UT3</th>\n",
       "      <th>base__so_UT5</th>\n",
       "      <th>repeat__repeatclass_LINE</th>\n",
       "      <th>repeat__repeatclass_LINE,SINE</th>\n",
       "      <th>repeat__repeatclass_LTR</th>\n",
       "      <th>repeat__repeatclass_Low_complexity</th>\n",
       "      <th>repeat__repeatclass_SINE</th>\n",
       "      <th>repeat__repeatclass_SINE,LINE</th>\n",
       "      <th>repeat__repeatclass_SINE,SINE</th>\n",
       "      <th>repeat__repeatclass_Satellite</th>\n",
       "      <th>repeat__repeatclass_Simple_repeat</th>\n",
       "      <th>repeat__repeatclass_Simple_repeat,SINE</th>\n",
       "      <th>repeat__repeatfamily_Alu</th>\n",
       "      <th>repeat__repeatfamily_Alu,Alu</th>\n",
       "      <th>repeat__repeatfamily_CR1</th>\n",
       "      <th>repeat__repeatfamily_ERV1</th>\n",
       "      <th>repeat__repeatfamily_ERVK</th>\n",
       "      <th>repeat__repeatfamily_ERVL</th>\n",
       "      <th>repeat__repeatfamily_ERVL-MaLR</th>\n",
       "      <th>repeat__repeatfamily_L1</th>\n",
       "      <th>repeat__repeatfamily_L1,Alu</th>\n",
       "      <th>repeat__repeatfamily_L2</th>\n",
       "      <th>repeat__repeatfamily_Low_complexity</th>\n",
       "      <th>repeat__repeatfamily_MIR</th>\n",
       "      <th>repeat__repeatfamily_MIR,L2</th>\n",
       "      <th>repeat__repeatfamily_RTE-BovB</th>\n",
       "      <th>repeat__repeatfamily_Satellite</th>\n",
       "      <th>repeat__repeatfamily_Simple_repeat</th>\n",
       "      <th>repeat__repeatfamily_Simple_repeat,Alu</th>\n",
       "      <th>repeat__repeatfamily_tRNA-RTE</th>\n",
       "      <th>esp6500__ea_pop_af</th>\n",
       "      <th>esp6500__aa_pop_af</th>\n",
       "      <th>gnomad__af</th>\n",
       "      <th>gnomad__af_afr</th>\n",
       "      <th>gnomad__af_amr</th>\n",
       "      <th>gnomad__af_asj</th>\n",
       "      <th>gnomad__af_eas</th>\n",
       "      <th>gnomad__af_fin</th>\n",
       "      <th>gnomad__af_nfe</th>\n",
       "      <th>gnomad__af_oth</th>\n",
       "      <th>gnomad__af_sas</th>\n",
       "      <th>uk10k_cohort__uk10k_twins_af</th>\n",
       "      <th>uk10k_cohort__uk10k_alspac_af</th>\n",
       "      <th>uk10k_cohort__uk10k_af</th>\n",
       "      <th>base__ref_base_len</th>\n",
       "      <th>base__alt_base_len</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000028</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000099</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000044</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000033</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000064</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000130</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000032</td>\n",
       "      <td>0.000115</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>0.000120</td>\n",
       "      <td>0.000028</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000032</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000642</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   base__coding_Y  base__transcript_1  base__so_2KD  base__so_2KU  \\\n",
       "0               1                   1             0             0   \n",
       "1               1                   1             0             0   \n",
       "2               1                   1             0             0   \n",
       "3               1                   1             0             0   \n",
       "4               1                   1             0             0   \n",
       "\n",
       "   base__so_CSS  base__so_FSD  base__so_IND  base__so_INT  base__so_MIS  \\\n",
       "0             0             0             0             0             0   \n",
       "1             0             0             0             0             1   \n",
       "2             0             0             0             0             1   \n",
       "3             0             0             0             0             0   \n",
       "4             0             0             0             0             1   \n",
       "\n",
       "   base__so_SPL  base__so_STG  base__so_STL  base__so_SYN  base__so_UNK  \\\n",
       "0             0             1             0             0             0   \n",
       "1             0             0             0             0             0   \n",
       "2             0             0             0             0             0   \n",
       "3             0             0             0             1             0   \n",
       "4             0             0             0             0             0   \n",
       "\n",
       "   base__so_UT3  base__so_UT5  repeat__repeatclass_LINE  \\\n",
       "0             0             0                         0   \n",
       "1             0             0                         0   \n",
       "2             0             0                         0   \n",
       "3             0             0                         0   \n",
       "4             0             0                         0   \n",
       "\n",
       "   repeat__repeatclass_LINE,SINE  repeat__repeatclass_LTR  \\\n",
       "0                              0                        0   \n",
       "1                              0                        0   \n",
       "2                              0                        0   \n",
       "3                              0                        0   \n",
       "4                              0                        0   \n",
       "\n",
       "   repeat__repeatclass_Low_complexity  repeat__repeatclass_SINE  \\\n",
       "0                                   0                         0   \n",
       "1                                   0                         0   \n",
       "2                                   0                         0   \n",
       "3                                   0                         0   \n",
       "4                                   0                         0   \n",
       "\n",
       "   repeat__repeatclass_SINE,LINE  repeat__repeatclass_SINE,SINE  \\\n",
       "0                              0                              0   \n",
       "1                              0                              0   \n",
       "2                              0                              0   \n",
       "3                              0                              0   \n",
       "4                              0                              0   \n",
       "\n",
       "   repeat__repeatclass_Satellite  repeat__repeatclass_Simple_repeat  \\\n",
       "0                              0                                  0   \n",
       "1                              0                                  0   \n",
       "2                              0                                  0   \n",
       "3                              0                                  0   \n",
       "4                              0                                  0   \n",
       "\n",
       "   repeat__repeatclass_Simple_repeat,SINE  repeat__repeatfamily_Alu  \\\n",
       "0                                       0                         0   \n",
       "1                                       0                         0   \n",
       "2                                       0                         0   \n",
       "3                                       0                         0   \n",
       "4                                       0                         0   \n",
       "\n",
       "   repeat__repeatfamily_Alu,Alu  repeat__repeatfamily_CR1  \\\n",
       "0                             0                         0   \n",
       "1                             0                         0   \n",
       "2                             0                         0   \n",
       "3                             0                         0   \n",
       "4                             0                         0   \n",
       "\n",
       "   repeat__repeatfamily_ERV1  repeat__repeatfamily_ERVK  \\\n",
       "0                          0                          0   \n",
       "1                          0                          0   \n",
       "2                          0                          0   \n",
       "3                          0                          0   \n",
       "4                          0                          0   \n",
       "\n",
       "   repeat__repeatfamily_ERVL  repeat__repeatfamily_ERVL-MaLR  \\\n",
       "0                          0                               0   \n",
       "1                          0                               0   \n",
       "2                          0                               0   \n",
       "3                          0                               0   \n",
       "4                          0                               0   \n",
       "\n",
       "   repeat__repeatfamily_L1  repeat__repeatfamily_L1,Alu  \\\n",
       "0                        0                            0   \n",
       "1                        0                            0   \n",
       "2                        0                            0   \n",
       "3                        0                            0   \n",
       "4                        0                            0   \n",
       "\n",
       "   repeat__repeatfamily_L2  repeat__repeatfamily_Low_complexity  \\\n",
       "0                        0                                    0   \n",
       "1                        0                                    0   \n",
       "2                        0                                    0   \n",
       "3                        0                                    0   \n",
       "4                        0                                    0   \n",
       "\n",
       "   repeat__repeatfamily_MIR  repeat__repeatfamily_MIR,L2  \\\n",
       "0                         0                            0   \n",
       "1                         0                            0   \n",
       "2                         0                            0   \n",
       "3                         0                            0   \n",
       "4                         0                            0   \n",
       "\n",
       "   repeat__repeatfamily_RTE-BovB  repeat__repeatfamily_Satellite  \\\n",
       "0                              0                               0   \n",
       "1                              0                               0   \n",
       "2                              0                               0   \n",
       "3                              0                               0   \n",
       "4                              0                               0   \n",
       "\n",
       "   repeat__repeatfamily_Simple_repeat  repeat__repeatfamily_Simple_repeat,Alu  \\\n",
       "0                                   0                                       0   \n",
       "1                                   0                                       0   \n",
       "2                                   0                                       0   \n",
       "3                                   0                                       0   \n",
       "4                                   0                                       0   \n",
       "\n",
       "   repeat__repeatfamily_tRNA-RTE  esp6500__ea_pop_af  esp6500__aa_pop_af  \\\n",
       "0                              0                 0.0                 0.0   \n",
       "1                              0                 0.0                 0.0   \n",
       "2                              0                 0.0                 0.0   \n",
       "3                              0                 0.0                 0.0   \n",
       "4                              0                 0.0                 0.0   \n",
       "\n",
       "   gnomad__af  gnomad__af_afr  gnomad__af_amr  gnomad__af_asj  gnomad__af_eas  \\\n",
       "0    0.000028        0.000000        0.000000        0.000099        0.000000   \n",
       "1    0.000064        0.000000        0.000000        0.000000        0.000000   \n",
       "2    0.000032        0.000115        0.000000        0.000000        0.000000   \n",
       "3    0.000014        0.000120        0.000028        0.000000        0.000000   \n",
       "4    0.000032        0.000000        0.000000        0.000000        0.000642   \n",
       "\n",
       "   gnomad__af_fin  gnomad__af_nfe  gnomad__af_oth  gnomad__af_sas  \\\n",
       "0             0.0        0.000044             0.0        0.000033   \n",
       "1             0.0        0.000130             0.0        0.000000   \n",
       "2             0.0        0.000000             0.0        0.000000   \n",
       "3             0.0        0.000000             0.0        0.000000   \n",
       "4             0.0        0.000000             0.0        0.000000   \n",
       "\n",
       "   uk10k_cohort__uk10k_twins_af  uk10k_cohort__uk10k_alspac_af  \\\n",
       "0                           0.0                            0.0   \n",
       "1                           0.0                            0.0   \n",
       "2                           0.0                            0.0   \n",
       "3                           0.0                            0.0   \n",
       "4                           0.0                            0.0   \n",
       "\n",
       "   uk10k_cohort__uk10k_af  base__ref_base_len  base__alt_base_len  \n",
       "0                     0.0                   1                   1  \n",
       "1                     0.0                   1                   1  \n",
       "2                     0.0                   1                   1  \n",
       "3                     0.0                   1                   1  \n",
       "4                     0.0                   1                   1  "
      ]
     },
     "execution_count": 215,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import re\n",
    "import pandas as pd\n",
    "import sqlite3\n",
    "conn = sqlite3.connect(\"../chr19_test_data/chr19_test.sqlite\")\n",
    "# cursor = conn.cursor()\n",
    "# cursor.execute(\"SELECT name FROM sqlite_master WHERE type='table';\")\n",
    "# print(cursor.fetchall())\n",
    "\n",
    "variant_annotator = pd.read_sql_query(\"SELECT * FROM variant_annotator\", conn)\n",
    "variant = pd.read_sql_query(\"SELECT * FROM variant\", conn)\n",
    "variant_header  = pd.read_sql_query(\"SELECT * FROM variant_header\", conn)\n",
    "variant_reportsub = pd.read_sql_query(\"SELECT * FROM variant_reportsub\", conn)\n",
    "gene_annotator = pd.read_sql_query(\"SELECT * FROM gene_annotator\", conn)\n",
    "gene = pd.read_sql_query(\"SELECT * FROM gene\", conn)\n",
    "gene_header  = pd.read_sql_query(\"SELECT * FROM gene_header\", conn)\n",
    "gene_reportsub = pd.read_sql_query(\"SELECT * FROM gene_reportsub\", conn)\n",
    "sample = pd.read_sql_query(\"SELECT * FROM sample\", conn)\n",
    "mapping  = pd.read_sql_query(\"SELECT * FROM mapping\", conn)\n",
    "variant_filtered = pd.read_sql_query(\"SELECT * FROM variant_filtered\", conn)\n",
    "gene_filtered = pd.read_sql_query(\"SELECT * FROM gene_filtered\", conn)\n",
    "info = pd.read_sql_query(\"SELECT * FROM info\", conn)\n",
    "\n",
    "# Objects are basically the non-numeric datatypes which we will need to convert. \n",
    "# variant.dtypes\n",
    "\n",
    "# variant.rownames = variant['dbsnp__snp']\n",
    "\n",
    "# These columns seem to have little/no significance as of now and will also increase the size of the matrix.\n",
    "dropCols = [\n",
    "    'base__chrom',\n",
    "    'base__pos',\n",
    "    'base__note',\n",
    "    'base__achange',\n",
    "    'base__hugo',\n",
    "    'base__all_mappings',\n",
    "    'vest__all_results',\n",
    "    'vest__hugo',\n",
    "    'vest__transcript',\n",
    "    'base__uid',\n",
    "    'denovo__PubmedId',\n",
    "    'denovo__Validation',\n",
    "    'denovo__PrimaryPhenotype',\n",
    "    'cosmic__protein_change',\n",
    "    'cosmic__transcript',\n",
    "    'cosmic__variant_count',\n",
    "    'cosmic__cosmic_id',\n",
    "    'clinvar__sig',\n",
    "    'clinvar__id',\n",
    "    'clinvar__rev_stat',\n",
    "    'clinvar__disease_refs',\n",
    "    'clinvar__disease_names',\n",
    "    'gtex__gtex_gene',\n",
    "    'gtex__gtex_tissue',\n",
    "    'uk10k_cohort__uk10k_twins_ac',\n",
    "    'uk10k_cohort__uk10k_alspac_ac',\n",
    "    'uk10k_cohort__uk10k_ac',\n",
    "    'vest__pval',\n",
    "    'tagsampler__numsample',\n",
    "    'tagsampler__samples',\n",
    "    'tagsampler__tags',\n",
    "    'repeat__repeatname',\n",
    "    'vest__score',\n",
    "    'dbsnp__snp',\n",
    "    'cosmic__variant_count_tissue'\n",
    "]\n",
    "\n",
    "# Dropping thousand genomes columns since they're all 0\n",
    "r = re.compile('^thousand*')\n",
    "dropCols.extend(list(filter(r.match,variant.columns)))\n",
    "variant = variant.drop(columns=dropCols, axis=1)\n",
    "\n",
    "# We will also assume that if there are no frequnecy values in frequency columns then we will assign a 0\n",
    "# Selecting all the allele frequency columns\n",
    "r = re.compile(\".*af.*\")\n",
    "afCols = list(filter(r.match,variant.columns))\n",
    "for col in afCols:\n",
    "    variant[col][variant[col].isnull()] = 0\n",
    "    variant[col].astype('float64')\n",
    "    \n",
    "# To further reduce featureset we will use the length of the variant as the ref base/alt base for now.\n",
    "# We can use the achange field's biochemical properties to provide further resolution in the absence\n",
    "# of sequence content.\n",
    "def getLen(x):\n",
    "    return len(x)\n",
    "\n",
    "# variant['base__ref_base_len'] = 0\n",
    "# variant['base__alt_base_len'] = 0\n",
    "variant['base__ref_base_len'] = variant['base__ref_base'].apply(getLen)\n",
    "variant['base__alt_base_len'] = variant['base__alt_base'].apply(getLen)\n",
    "variant['base__ref_base_len'].astype('int64')\n",
    "variant['base__alt_base_len'].astype('int64')\n",
    "variant = variant.drop(columns=['base__ref_base','base__alt_base'], axis=1)\n",
    "\n",
    "\n",
    "# We will assume that if there is a transcript ID then then variant will be in an exon so we will set this as a binary\n",
    "# flag.\n",
    "variant['base__transcript'][variant['base__transcript'].isnull()] = 0\n",
    "variant['base__transcript'][variant['base__transcript'].notnull()] = 1\n",
    "\n",
    "#  selecting only the 'object' dtypes (categorigal variables). \n",
    "cat_var = variant.select_dtypes(include=['object']).copy()\n",
    "con_var = variant.select_dtypes(exclude=['object']).copy()\n",
    "\n",
    "# This is label encoding \n",
    "# which seems like it might not be great for clustering, \n",
    "# though not sure for hierarchical clustering...\n",
    "# for col in test_var.columns:\n",
    "#     test_var[col] = test_var[col].astype('category')\n",
    "#     test_var[col+'_cat'] = test_var[col].cat.codes\n",
    "\n",
    "# This is how to do one-hot encoding. Very similar to the melt fxn in R'\n",
    "test = pd.get_dummies(cat_var)\n",
    "\n",
    "var = pd.concat([test, con_var], sort=False, axis=1)\n",
    "var.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 216,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(10096, 23)\n",
      "(10096, 16)\n",
      "(10096, 7)\n",
      "(10096, 44)\n",
      "(10096, 60)\n"
     ]
    }
   ],
   "source": [
    "print(variant.shape)\n",
    "print(con_var.shape)\n",
    "print(cat_var.shape)\n",
    "print(test.shape)\n",
    "print(var.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 183,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1    9887\n",
       "2      85\n",
       "3      36\n",
       "4      36\n",
       "9      28\n",
       "5      10\n",
       "6       9\n",
       "7       4\n",
       "8       1\n",
       "Name: base__alt_base_len, dtype: int64"
      ]
     },
     "execution_count": 183,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# variant.select_dtypes(include=['object'])\n",
    "# variant.dtypes\n",
    "variant['base__alt_base_len'].value_counts().sort_values(ascending=False)\n",
    "# variant[afCols].isnull()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Joining Datasets\n",
    "The file linking everything together will be the annotation data from NG00061 as it should hold the complete set of SNPs. I will use coordinates from this table to join."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "headers = ['chr','pos']\n",
    "temp = anno.merge(pvalue, how='left', left_on=headers, right_on=headers)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>alt_allele</th>\n",
       "      <th>seq_meta_var_id</th>\n",
       "      <th>epacts_var_id</th>\n",
       "      <th>Allele1</th>\n",
       "      <th>Allele2</th>\n",
       "      <th>P-value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10</td>\n",
       "      <td>60494</td>\n",
       "      <td>G</td>\n",
       "      <td>10:60494A,G</td>\n",
       "      <td>10:60494_A/G</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>0.32420</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10</td>\n",
       "      <td>60523</td>\n",
       "      <td>G</td>\n",
       "      <td>10:60523T,G</td>\n",
       "      <td>10:60523_T/G</td>\n",
       "      <td>t</td>\n",
       "      <td>g</td>\n",
       "      <td>0.37210</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10</td>\n",
       "      <td>61331</td>\n",
       "      <td>G</td>\n",
       "      <td>10:61331A,G</td>\n",
       "      <td>10:61331_A/G</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>0.75160</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10</td>\n",
       "      <td>61334</td>\n",
       "      <td>A</td>\n",
       "      <td>10:61334G,A</td>\n",
       "      <td>10:61334_G/A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>10</td>\n",
       "      <td>61646</td>\n",
       "      <td>G</td>\n",
       "      <td>10:61646A,G</td>\n",
       "      <td>10:61646_A/G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>10</td>\n",
       "      <td>61654</td>\n",
       "      <td>A</td>\n",
       "      <td>10:61654G,A</td>\n",
       "      <td>10:61654_G/A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>10</td>\n",
       "      <td>61766</td>\n",
       "      <td>A</td>\n",
       "      <td>10:61766G,A</td>\n",
       "      <td>10:61766_G/A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>10</td>\n",
       "      <td>62450</td>\n",
       "      <td>A</td>\n",
       "      <td>10:62450G,A</td>\n",
       "      <td>10:62450_G/A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>10</td>\n",
       "      <td>66295</td>\n",
       "      <td>T</td>\n",
       "      <td>10:66295C,T</td>\n",
       "      <td>10:66295_C/T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>10</td>\n",
       "      <td>66326</td>\n",
       "      <td>G</td>\n",
       "      <td>10:66326A,G</td>\n",
       "      <td>10:66326_A/G</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>0.58810</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>10</td>\n",
       "      <td>67774</td>\n",
       "      <td>C</td>\n",
       "      <td>10:67774G,C</td>\n",
       "      <td>10:67774_G/C</td>\n",
       "      <td>c</td>\n",
       "      <td>g</td>\n",
       "      <td>0.24030</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>10</td>\n",
       "      <td>69083</td>\n",
       "      <td>T</td>\n",
       "      <td>10:69083C,T</td>\n",
       "      <td>10:69083_C/T</td>\n",
       "      <td>t</td>\n",
       "      <td>c</td>\n",
       "      <td>0.18780</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>10</td>\n",
       "      <td>69123</td>\n",
       "      <td>A</td>\n",
       "      <td>10:69123G,A</td>\n",
       "      <td>10:69123_G/A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>10</td>\n",
       "      <td>71718</td>\n",
       "      <td>A</td>\n",
       "      <td>10:71718T,A</td>\n",
       "      <td>10:71718_T/A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>10</td>\n",
       "      <td>71776</td>\n",
       "      <td>C</td>\n",
       "      <td>10:71776T,C</td>\n",
       "      <td>10:71776_T/C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>10</td>\n",
       "      <td>73496</td>\n",
       "      <td>T</td>\n",
       "      <td>10:73496C,T</td>\n",
       "      <td>10:73496_C/T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>10</td>\n",
       "      <td>73537</td>\n",
       "      <td>A</td>\n",
       "      <td>10:73537G,A</td>\n",
       "      <td>10:73537_G/A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>10</td>\n",
       "      <td>73704</td>\n",
       "      <td>T</td>\n",
       "      <td>10:73704C,T</td>\n",
       "      <td>10:73704_C/T</td>\n",
       "      <td>t</td>\n",
       "      <td>c</td>\n",
       "      <td>0.20890</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>10</td>\n",
       "      <td>73759</td>\n",
       "      <td>T</td>\n",
       "      <td>10:73759C,T</td>\n",
       "      <td>10:73759_C/T</td>\n",
       "      <td>t</td>\n",
       "      <td>c</td>\n",
       "      <td>0.27150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>10</td>\n",
       "      <td>74504</td>\n",
       "      <td>A</td>\n",
       "      <td>10:74504G,A</td>\n",
       "      <td>10:74504_G/A</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>0.45970</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>10</td>\n",
       "      <td>74549</td>\n",
       "      <td>T</td>\n",
       "      <td>10:74549G,T</td>\n",
       "      <td>10:74549_G/T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>10</td>\n",
       "      <td>74568</td>\n",
       "      <td>A</td>\n",
       "      <td>10:74568T,A</td>\n",
       "      <td>10:74568_T/A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>10</td>\n",
       "      <td>74572</td>\n",
       "      <td>C</td>\n",
       "      <td>10:74572G,C</td>\n",
       "      <td>10:74572_G/C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>10</td>\n",
       "      <td>75552</td>\n",
       "      <td>A</td>\n",
       "      <td>10:75552C,A</td>\n",
       "      <td>10:75552_C/A</td>\n",
       "      <td>a</td>\n",
       "      <td>c</td>\n",
       "      <td>0.41880</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>10</td>\n",
       "      <td>75794</td>\n",
       "      <td>T</td>\n",
       "      <td>10:75794G,T</td>\n",
       "      <td>10:75794_G/T</td>\n",
       "      <td>t</td>\n",
       "      <td>g</td>\n",
       "      <td>0.23190</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>10</td>\n",
       "      <td>75800</td>\n",
       "      <td>T</td>\n",
       "      <td>10:75800C,T</td>\n",
       "      <td>10:75800_C/T</td>\n",
       "      <td>t</td>\n",
       "      <td>c</td>\n",
       "      <td>0.01283</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>10</td>\n",
       "      <td>75803</td>\n",
       "      <td>C</td>\n",
       "      <td>10:75803T,C</td>\n",
       "      <td>10:75803_T/C</td>\n",
       "      <td>t</td>\n",
       "      <td>c</td>\n",
       "      <td>0.32660</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>10</td>\n",
       "      <td>76210</td>\n",
       "      <td>G</td>\n",
       "      <td>10:76210A,G</td>\n",
       "      <td>10:76210_A/G</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>0.04852</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>10</td>\n",
       "      <td>76294</td>\n",
       "      <td>C</td>\n",
       "      <td>10:76294T,C</td>\n",
       "      <td>10:76294_T/C</td>\n",
       "      <td>t</td>\n",
       "      <td>c</td>\n",
       "      <td>0.36500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>10</td>\n",
       "      <td>77582</td>\n",
       "      <td>G</td>\n",
       "      <td>10:77582A,G</td>\n",
       "      <td>10:77582_A/G</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>0.05238</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24832767</th>\n",
       "      <td>7</td>\n",
       "      <td>134128708</td>\n",
       "      <td>T</td>\n",
       "      <td>7:134128708C,T</td>\n",
       "      <td>7:134128708_C/T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24832768</th>\n",
       "      <td>7</td>\n",
       "      <td>134128725</td>\n",
       "      <td>A</td>\n",
       "      <td>7:134128725C,A</td>\n",
       "      <td>7:134128725_C/A</td>\n",
       "      <td>a</td>\n",
       "      <td>c</td>\n",
       "      <td>0.88780</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24832769</th>\n",
       "      <td>7</td>\n",
       "      <td>134128816</td>\n",
       "      <td>A</td>\n",
       "      <td>7:134128816G,A</td>\n",
       "      <td>7:134128816_G/A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24832770</th>\n",
       "      <td>7</td>\n",
       "      <td>134129006</td>\n",
       "      <td>G</td>\n",
       "      <td>7:134129006T,G</td>\n",
       "      <td>7:134129006_T/G</td>\n",
       "      <td>t</td>\n",
       "      <td>g</td>\n",
       "      <td>0.59260</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24832771</th>\n",
       "      <td>7</td>\n",
       "      <td>134129017</td>\n",
       "      <td>T</td>\n",
       "      <td>7:134129017C,T</td>\n",
       "      <td>7:134129017_C/T</td>\n",
       "      <td>t</td>\n",
       "      <td>c</td>\n",
       "      <td>0.55110</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24832772</th>\n",
       "      <td>7</td>\n",
       "      <td>134129165</td>\n",
       "      <td>A</td>\n",
       "      <td>7:134129165C,A</td>\n",
       "      <td>7:134129165_C/A</td>\n",
       "      <td>a</td>\n",
       "      <td>c</td>\n",
       "      <td>0.89970</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24832773</th>\n",
       "      <td>7</td>\n",
       "      <td>134129178</td>\n",
       "      <td>A</td>\n",
       "      <td>7:134129178G,A</td>\n",
       "      <td>7:134129178_G/A</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>0.88790</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24832774</th>\n",
       "      <td>7</td>\n",
       "      <td>134129323</td>\n",
       "      <td>G</td>\n",
       "      <td>7:134129323A,G</td>\n",
       "      <td>7:134129323_A/G</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>0.05273</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24832775</th>\n",
       "      <td>7</td>\n",
       "      <td>134129346</td>\n",
       "      <td>A</td>\n",
       "      <td>7:134129346G,A</td>\n",
       "      <td>7:134129346_G/A</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>0.94760</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24832776</th>\n",
       "      <td>7</td>\n",
       "      <td>134129384</td>\n",
       "      <td>T</td>\n",
       "      <td>7:134129384C,T</td>\n",
       "      <td>7:134129384_C/T</td>\n",
       "      <td>t</td>\n",
       "      <td>c</td>\n",
       "      <td>0.86980</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24832777</th>\n",
       "      <td>7</td>\n",
       "      <td>134129529</td>\n",
       "      <td>A</td>\n",
       "      <td>7:134129529G,A</td>\n",
       "      <td>7:134129529_G/A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24832778</th>\n",
       "      <td>7</td>\n",
       "      <td>134129660</td>\n",
       "      <td>C</td>\n",
       "      <td>7:134129660A,C</td>\n",
       "      <td>7:134129660_A/C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24832779</th>\n",
       "      <td>7</td>\n",
       "      <td>134129901</td>\n",
       "      <td>C</td>\n",
       "      <td>7:134129901T,C</td>\n",
       "      <td>7:134129901_T/C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24832780</th>\n",
       "      <td>7</td>\n",
       "      <td>134130168</td>\n",
       "      <td>C</td>\n",
       "      <td>7:134130168G,C</td>\n",
       "      <td>7:134130168_G/C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24832781</th>\n",
       "      <td>7</td>\n",
       "      <td>134130177</td>\n",
       "      <td>A</td>\n",
       "      <td>7:134130177G,A</td>\n",
       "      <td>7:134130177_G/A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24832782</th>\n",
       "      <td>7</td>\n",
       "      <td>134130563</td>\n",
       "      <td>A</td>\n",
       "      <td>7:134130563C,A</td>\n",
       "      <td>7:134130563_C/A</td>\n",
       "      <td>a</td>\n",
       "      <td>c</td>\n",
       "      <td>0.29020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24832783</th>\n",
       "      <td>7</td>\n",
       "      <td>134130669</td>\n",
       "      <td>G</td>\n",
       "      <td>7:134130669T,G</td>\n",
       "      <td>7:134130669_T/G</td>\n",
       "      <td>t</td>\n",
       "      <td>g</td>\n",
       "      <td>0.05620</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24832784</th>\n",
       "      <td>7</td>\n",
       "      <td>134130673</td>\n",
       "      <td>C</td>\n",
       "      <td>7:134130673G,C</td>\n",
       "      <td>7:134130673_G/C</td>\n",
       "      <td>c</td>\n",
       "      <td>g</td>\n",
       "      <td>0.29240</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24832785</th>\n",
       "      <td>7</td>\n",
       "      <td>134130702</td>\n",
       "      <td>T</td>\n",
       "      <td>7:134130702C,T</td>\n",
       "      <td>7:134130702_C/T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24832786</th>\n",
       "      <td>7</td>\n",
       "      <td>134130720</td>\n",
       "      <td>G</td>\n",
       "      <td>7:134130720A,G</td>\n",
       "      <td>7:134130720_A/G</td>\n",
       "      <td>a</td>\n",
       "      <td>g</td>\n",
       "      <td>0.16850</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24832787</th>\n",
       "      <td>7</td>\n",
       "      <td>134130731</td>\n",
       "      <td>T</td>\n",
       "      <td>7:134130731C,T</td>\n",
       "      <td>7:134130731_C/T</td>\n",
       "      <td>t</td>\n",
       "      <td>c</td>\n",
       "      <td>0.06880</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24832788</th>\n",
       "      <td>7</td>\n",
       "      <td>134130828</td>\n",
       "      <td>G</td>\n",
       "      <td>7:134130828T,G</td>\n",
       "      <td>7:134130828_T/G</td>\n",
       "      <td>t</td>\n",
       "      <td>g</td>\n",
       "      <td>0.17730</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24832789</th>\n",
       "      <td>7</td>\n",
       "      <td>134130891</td>\n",
       "      <td>A</td>\n",
       "      <td>7:134130891C,A</td>\n",
       "      <td>7:134130891_C/A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24832790</th>\n",
       "      <td>7</td>\n",
       "      <td>134131218</td>\n",
       "      <td>T</td>\n",
       "      <td>7:134131218C,T</td>\n",
       "      <td>7:134131218_C/T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24832791</th>\n",
       "      <td>7</td>\n",
       "      <td>134131222</td>\n",
       "      <td>T</td>\n",
       "      <td>7:134131222C,T</td>\n",
       "      <td>7:134131222_C/T</td>\n",
       "      <td>t</td>\n",
       "      <td>c</td>\n",
       "      <td>0.05575</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24832792</th>\n",
       "      <td>7</td>\n",
       "      <td>134131251</td>\n",
       "      <td>A</td>\n",
       "      <td>7:134131251T,A</td>\n",
       "      <td>7:134131251_T/A</td>\n",
       "      <td>a</td>\n",
       "      <td>t</td>\n",
       "      <td>0.88970</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24832793</th>\n",
       "      <td>7</td>\n",
       "      <td>134131343</td>\n",
       "      <td>C</td>\n",
       "      <td>7:134131343T,C</td>\n",
       "      <td>7:134131343_T/C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24832794</th>\n",
       "      <td>7</td>\n",
       "      <td>134131408</td>\n",
       "      <td>A</td>\n",
       "      <td>7:134131408G,A</td>\n",
       "      <td>7:134131408_G/A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24832795</th>\n",
       "      <td>7</td>\n",
       "      <td>134131456</td>\n",
       "      <td>C</td>\n",
       "      <td>7:134131456G,C</td>\n",
       "      <td>7:134131456_G/C</td>\n",
       "      <td>c</td>\n",
       "      <td>g</td>\n",
       "      <td>0.15940</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24832796</th>\n",
       "      <td>7</td>\n",
       "      <td>134131907</td>\n",
       "      <td>T</td>\n",
       "      <td>7:134131907C,T</td>\n",
       "      <td>7:134131907_C/T</td>\n",
       "      <td>t</td>\n",
       "      <td>c</td>\n",
       "      <td>0.89000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>24832797 rows  8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          chr        pos alt_allele seq_meta_var_id    epacts_var_id Allele1  \\\n",
       "0          10      60494          G     10:60494A,G     10:60494_A/G       a   \n",
       "1          10      60523          G     10:60523T,G     10:60523_T/G       t   \n",
       "2          10      61331          G     10:61331A,G     10:61331_A/G       a   \n",
       "3          10      61334          A     10:61334G,A     10:61334_G/A     NaN   \n",
       "4          10      61646          G     10:61646A,G     10:61646_A/G     NaN   \n",
       "5          10      61654          A     10:61654G,A     10:61654_G/A     NaN   \n",
       "6          10      61766          A     10:61766G,A     10:61766_G/A     NaN   \n",
       "7          10      62450          A     10:62450G,A     10:62450_G/A     NaN   \n",
       "8          10      66295          T     10:66295C,T     10:66295_C/T     NaN   \n",
       "9          10      66326          G     10:66326A,G     10:66326_A/G       a   \n",
       "10         10      67774          C     10:67774G,C     10:67774_G/C       c   \n",
       "11         10      69083          T     10:69083C,T     10:69083_C/T       t   \n",
       "12         10      69123          A     10:69123G,A     10:69123_G/A     NaN   \n",
       "13         10      71718          A     10:71718T,A     10:71718_T/A     NaN   \n",
       "14         10      71776          C     10:71776T,C     10:71776_T/C     NaN   \n",
       "15         10      73496          T     10:73496C,T     10:73496_C/T     NaN   \n",
       "16         10      73537          A     10:73537G,A     10:73537_G/A     NaN   \n",
       "17         10      73704          T     10:73704C,T     10:73704_C/T       t   \n",
       "18         10      73759          T     10:73759C,T     10:73759_C/T       t   \n",
       "19         10      74504          A     10:74504G,A     10:74504_G/A       a   \n",
       "20         10      74549          T     10:74549G,T     10:74549_G/T     NaN   \n",
       "21         10      74568          A     10:74568T,A     10:74568_T/A     NaN   \n",
       "22         10      74572          C     10:74572G,C     10:74572_G/C     NaN   \n",
       "23         10      75552          A     10:75552C,A     10:75552_C/A       a   \n",
       "24         10      75794          T     10:75794G,T     10:75794_G/T       t   \n",
       "25         10      75800          T     10:75800C,T     10:75800_C/T       t   \n",
       "26         10      75803          C     10:75803T,C     10:75803_T/C       t   \n",
       "27         10      76210          G     10:76210A,G     10:76210_A/G       a   \n",
       "28         10      76294          C     10:76294T,C     10:76294_T/C       t   \n",
       "29         10      77582          G     10:77582A,G     10:77582_A/G       a   \n",
       "...       ...        ...        ...             ...              ...     ...   \n",
       "24832767    7  134128708          T  7:134128708C,T  7:134128708_C/T     NaN   \n",
       "24832768    7  134128725          A  7:134128725C,A  7:134128725_C/A       a   \n",
       "24832769    7  134128816          A  7:134128816G,A  7:134128816_G/A     NaN   \n",
       "24832770    7  134129006          G  7:134129006T,G  7:134129006_T/G       t   \n",
       "24832771    7  134129017          T  7:134129017C,T  7:134129017_C/T       t   \n",
       "24832772    7  134129165          A  7:134129165C,A  7:134129165_C/A       a   \n",
       "24832773    7  134129178          A  7:134129178G,A  7:134129178_G/A       a   \n",
       "24832774    7  134129323          G  7:134129323A,G  7:134129323_A/G       a   \n",
       "24832775    7  134129346          A  7:134129346G,A  7:134129346_G/A       a   \n",
       "24832776    7  134129384          T  7:134129384C,T  7:134129384_C/T       t   \n",
       "24832777    7  134129529          A  7:134129529G,A  7:134129529_G/A     NaN   \n",
       "24832778    7  134129660          C  7:134129660A,C  7:134129660_A/C     NaN   \n",
       "24832779    7  134129901          C  7:134129901T,C  7:134129901_T/C     NaN   \n",
       "24832780    7  134130168          C  7:134130168G,C  7:134130168_G/C     NaN   \n",
       "24832781    7  134130177          A  7:134130177G,A  7:134130177_G/A     NaN   \n",
       "24832782    7  134130563          A  7:134130563C,A  7:134130563_C/A       a   \n",
       "24832783    7  134130669          G  7:134130669T,G  7:134130669_T/G       t   \n",
       "24832784    7  134130673          C  7:134130673G,C  7:134130673_G/C       c   \n",
       "24832785    7  134130702          T  7:134130702C,T  7:134130702_C/T     NaN   \n",
       "24832786    7  134130720          G  7:134130720A,G  7:134130720_A/G       a   \n",
       "24832787    7  134130731          T  7:134130731C,T  7:134130731_C/T       t   \n",
       "24832788    7  134130828          G  7:134130828T,G  7:134130828_T/G       t   \n",
       "24832789    7  134130891          A  7:134130891C,A  7:134130891_C/A     NaN   \n",
       "24832790    7  134131218          T  7:134131218C,T  7:134131218_C/T     NaN   \n",
       "24832791    7  134131222          T  7:134131222C,T  7:134131222_C/T       t   \n",
       "24832792    7  134131251          A  7:134131251T,A  7:134131251_T/A       a   \n",
       "24832793    7  134131343          C  7:134131343T,C  7:134131343_T/C     NaN   \n",
       "24832794    7  134131408          A  7:134131408G,A  7:134131408_G/A     NaN   \n",
       "24832795    7  134131456          C  7:134131456G,C  7:134131456_G/C       c   \n",
       "24832796    7  134131907          T  7:134131907C,T  7:134131907_C/T       t   \n",
       "\n",
       "         Allele2  P-value  \n",
       "0              g  0.32420  \n",
       "1              g  0.37210  \n",
       "2              g  0.75160  \n",
       "3            NaN      NaN  \n",
       "4            NaN      NaN  \n",
       "5            NaN      NaN  \n",
       "6            NaN      NaN  \n",
       "7            NaN      NaN  \n",
       "8            NaN      NaN  \n",
       "9              g  0.58810  \n",
       "10             g  0.24030  \n",
       "11             c  0.18780  \n",
       "12           NaN      NaN  \n",
       "13           NaN      NaN  \n",
       "14           NaN      NaN  \n",
       "15           NaN      NaN  \n",
       "16           NaN      NaN  \n",
       "17             c  0.20890  \n",
       "18             c  0.27150  \n",
       "19             g  0.45970  \n",
       "20           NaN      NaN  \n",
       "21           NaN      NaN  \n",
       "22           NaN      NaN  \n",
       "23             c  0.41880  \n",
       "24             g  0.23190  \n",
       "25             c  0.01283  \n",
       "26             c  0.32660  \n",
       "27             g  0.04852  \n",
       "28             c  0.36500  \n",
       "29             g  0.05238  \n",
       "...          ...      ...  \n",
       "24832767     NaN      NaN  \n",
       "24832768       c  0.88780  \n",
       "24832769     NaN      NaN  \n",
       "24832770       g  0.59260  \n",
       "24832771       c  0.55110  \n",
       "24832772       c  0.89970  \n",
       "24832773       g  0.88790  \n",
       "24832774       g  0.05273  \n",
       "24832775       g  0.94760  \n",
       "24832776       c  0.86980  \n",
       "24832777     NaN      NaN  \n",
       "24832778     NaN      NaN  \n",
       "24832779     NaN      NaN  \n",
       "24832780     NaN      NaN  \n",
       "24832781     NaN      NaN  \n",
       "24832782       c  0.29020  \n",
       "24832783       g  0.05620  \n",
       "24832784       g  0.29240  \n",
       "24832785     NaN      NaN  \n",
       "24832786       g  0.16850  \n",
       "24832787       c  0.06880  \n",
       "24832788       g  0.17730  \n",
       "24832789     NaN      NaN  \n",
       "24832790     NaN      NaN  \n",
       "24832791       c  0.05575  \n",
       "24832792       t  0.88970  \n",
       "24832793     NaN      NaN  \n",
       "24832794     NaN      NaN  \n",
       "24832795       g  0.15940  \n",
       "24832796       c  0.89000  \n",
       "\n",
       "[24832797 rows x 8 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "temp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Breaks kernel. Don't run in notebook.\n",
    "from functools import reduce\n",
    "dataframes = [\n",
    "    anno,\n",
    "    pvalue,\n",
    "    conseq,\n",
    "    igap1,\n",
    "    igap12,\n",
    "    hapCerAd,\n",
    "    hapTxAd,\n",
    "    hapmapCerAd,\n",
    "    hapmapTxAd\n",
    "]\n",
    "df_merged = reduce(lambda  left,right: pd.merge(left,right,on=['chr','pos'],\n",
    "                                                how='outer'), dataframes).fillna('void')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  },
  "latex_envs": {
   "bibliofile": "biblio.bib",
   "cite_by": "apalike",
   "current_citInitial": 1,
   "eqLabelWithNumbers": true,
   "eqNumInitial": 0
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
